crates/router/src/services/authentication.rs

```diff
@@ -696,6 +696,348 @@ where
+                Err(e) => {
+                    logger::warn!("Admin API Auth failed: {:?}", e);
+                }
+            }
+        }
+
+        // Step 2: Try Platform Auth
+        let api_key = get_api_key(request_headers)
+            .change_context(errors::ApiErrorResponse::Unauthorized)?
+            .trim();
+        if api_key.is_empty() {
+            return Err(errors::ApiErrorResponse::Unauthorized)
+                .attach_printable("API key is empty");
+        }
+
+        let api_key_plaintext = api_keys::PlaintextApiKey::from(api_key);
+        let hash_key = state.conf().api_keys.get_inner().get_hash_key()?;
+        let hashed_api_key = api_key_plaintext.keyed_hash(hash_key.peek());
+
+        let stored_api_key = state
+            .store()
+            .find_api_key_by_hash_optional(hashed_api_key.into())
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to retrieve API key")?
+            .ok_or_else(|| report!(errors::ApiErrorResponse::Unauthorized))
+            .attach_printable("Merchant not authenticated via API key")?;
+
+        if stored_api_key
+            .expires_at
+            .map(|expires_at| expires_at < date_time::now())
+            .unwrap_or(false)
+        {
+            return Err(report!(errors::ApiErrorResponse::Unauthorized))
+                .attach_printable("API key has expired");
+        }
+
+        let key_manager_state = &(&state.session_state()).into();
+
+        let key_store = state
+            .store()
+            .get_merchant_key_store_by_merchant_id(
+                key_manager_state,
+                &stored_api_key.merchant_id,
+                &state.store().get_master_key().to_vec().into(),
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::Unauthorized)
+            .attach_printable("Failed to fetch merchant key store for the merchant id")?;
+
+        let merchant_account = state
+            .store()
+            .find_merchant_account_by_merchant_id(
+                key_manager_state,
+                &stored_api_key.merchant_id,
+                &key_store,
+            )
+            .await
+            .to_not_found_response(errors::ApiErrorResponse::Unauthorized)
+            .attach_printable("Merchant account not found")?;
+
+        if !(state.conf().platform.enabled
+            && merchant_account.merchant_account_type == MerchantAccountType::Platform)
+        {
+            return Err(report!(errors::ApiErrorResponse::Unauthorized)
+                .attach_printable("Platform authentication check failed"));
+        }
+
+        if let Some(organization_id) = self.organization_id.clone() {
+            if organization_id != merchant_account.get_org_id().clone() {
+                return Err(report!(errors::ApiErrorResponse::Unauthorized))
+                    .attach_printable("Organization ID does not match");
+            }
```

Comment:
Nit: Can probably use references here to avoid clones.

Something like this maybe:
```suggestion
        if let Some(ref organization_id) = self.organization_id {
            if organization_id != merchant_account.get_org_id() {
                return Err(report!(errors::ApiErrorResponse::Unauthorized))
                    .attach_printable("Organization ID does not match");
            }
```